projects/kserve/roles/kserve_deploy_model/templates/caikit-tgis-config.yaml.j2

```diff
@@ -23,5 +23,8 @@ model_management:
           config:
             local:
               load_timeout: 5000 # only relevant with the 1-container architecture
+{% if 'NUM_GPUS' in kserve_deploy_model_sr_kserve_extra_env_values %}
+              num_gpus: {{ kserve_deploy_model_sr_kserve_extra_env_values['NUM_GPUS'] }}
+{% endif %}
```

Comment:
I'm not 100% sure about this one vs checking for 
```
{% if kserve_deploy_model_sr_kserve_resource_request['nvidia.com/gpu'] %}
      num_gpus: {{  kserve_deploy_model_sr_kserve_resource_request['nvidia.com/gpu']  }}
{% endif %}
```
and same for setting `NUM_GPUS`
(I had this in some of the first versions of the serving runtime, but then I found out it was useless for 1 GPU -- at least)

Reply:
Hmmm, I wanted to avoid a keyerror but after experimenting with jinja2 it looks like this will be handled fine as you wrote it.

I also wanted to avoid using `nvidia.com/gpu` previously, as we were using gigabytes of GPU memory as a way to share the GPUs but since I removed this, I will go with your suggestion!